src/main/java/seedu/address/logic/parser/DeleteCommandParser.java

```diff
@@ -21,9 +25,29 @@ public DeleteCommand parse(String args) throws ParseException {
             Index index = ParserUtil.parseIndex(args);
             return new DeleteCommand(index);
         } catch (ParseException pe) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_USAGE), pe);
+            ArgumentMultimap argMultimap = checkValidity(args);
+
+            try {
+                String str = argMultimap.getValue(PREFIX_NAME).orElseThrow(() -> new Exception("shouldnt be thrown"));
+                Name name = ParserUtil.parseName(str);
+                return new DeleteCommand(name);
+            } catch (Exception ex) {
+                throw new ParseException(
+                        String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_USAGE), ex);
+            }
         }
     }
 
+    private ArgumentMultimap checkValidity(String args) throws ParseException {
```

Comment:
Would this be better to be `getArgMultimapByArgs` since `checkXXX` sounds that it would return a `boolean`.

Reply:
Yes, good catch!